### PR TITLE
Refactor recover command

### PR DESCRIPTION
**Pull request by tsudmi**

### CODE REVIEW

File: src/common/contracts.py [hunk 1]
```diff
@@ -92,6 +95,36 @@ async def get_validators_index(self) -> int:
...
+        return [Web3.to_hex(event['args']['publicKey']) for event in events]
+
+
+class V2PoolContract(ContractWrapper):
+    abi_path = 'abi/IV2Pool.json'
+    settings_key = 'V2_POOL_CONTRACT_ADDRESS'
+
+    async def get_registered_validators_public_keys(self) -> list[HexStr]:
+        """Fetches the validator registered events."""
+        blocks_per_month = int(SECONDS_PER_MONTH // settings.network_config.SECONDS_PER_BLOCK)
```
Comment (cyc60) [inline src/common/contracts.py:114]: Add comment pls

File: src/common/contracts.py [hunk 2]
```diff
@@ -92,6 +95,36 @@ async def get_validators_index(self) -> int:
...
+        to_block = BlockNumber(
+            min(
+                settings.network_config.KEEPER_GENESIS_BLOCK + blocks_per_month,
+                await execution_client.eth.get_block_number(),
+            )
+        )
```
Comment (evgeny-stakewise) [inline src/common/contracts.py:120]: Business logic should be in `_fetch_registered_validators`, not here. Contract  method should just take params as is or use simple defaults.

File: src/commands/recover.py [hunk 3]
```diff
@@ -122,106 +125,116 @@ async def main(
...
 
-    validators_status = {}
+    if vault_contract.contract_address == settings.network_config.GENESIS_VAULT_CONTRACT_ADDRESS:
+        public_keys.extend(await v2_pool_contract.get_registered_validators_public_keys())
+    click.secho(f'Fetched {len(public_keys)} registered validators', bold=True)
 
+    click.secho('Fetching validators statuses...', bold=True)
+    validator_statuses: dict[HexStr, ValidatorStatus | None] = {
+        public_key: None for public_key in public_keys
+    }
```
Comment (evgeny-stakewise) [inline src/commands/recover.py:174]: why dont you start with 
```validator_statuses: dict[HexStr, ValidatorStatus] = {} ``` 
?

Comment (tsudmi) [reply src/commands/recover.py:193]: Because we need public keys that are not yet in the beacon chain (they appear there after 1024 blocks)

**Review by cyc60 — APPROVED** [review]

**Review by evgeny-stakewise — APPROVED** [review]

**Pull request merged**